### PR TITLE
Add support for Jira Issue handler

### PR DESCRIPTION
**Pull request by jameerpathan111**
### Problem Statement
-  Missing support for Jira issue handling in Robottelo.

### Solution
- Add `BlockedBy` doc field: If the issue is open, the test will be skipped/deselected. Override behavior with `--blocked-by false`.
- Add `Verifies` doc field: Used with `--verifies-issues` pytest option.
- `--verifies-issues`: Comma-separated list of Jiras to collect tests matching the `Verifies` testimony marker. If no issue is provided, all tests with the `Verifies` testimony marker will be selected.
- `--blocked-by`: Comma-separated list of Jiras to collect tests matching the `BlockedBy` testimony marker. If no issue is provided, all tests with the `BlockedBy` testimony marker will be processed and deselected if any issue is open.
- `is_open_jira`: Function to check if a specific Jira is open.
### Related Issues
- SAT-22761

### CODE REVIEW

File: conf/jira.yaml.template [hunk 1]
```diff
@@ -0,0 +1,5 @@
+JIRA:
```
Comment (shweta83) [inline conf/jira.yaml.template:1]: Just a thought here, Can we merge both bugzilla.yaml and jira.yaml to one config file?

Comment (jameerpathan111) [reply conf/jira.yaml.template:1]: We plan to remove the code for bugzilla once we stop using it so I'd rather keep things separate for simplicity.

File: pytest_plugins/metadata_markers.py [hunk 2]
```diff
@@ -7,9 +7,22 @@
...
+deselected = []
+
+
+def parse_comma_separated_list(option_value):
+    if isinstance(option_value, str):
+        if option_value.lower() == 'true':
+            return True
+        if option_value.lower() == 'false':
+            return False
+        return option_value.split(',')
```
Comment (omkarkhatavkar) [inline pytest_plugins/metadata_markers.py:24]: return [item.strip() for item in option_value.split(',')]

just to handle this case 
--blocked-by '123, 2333,  322'

File: pytest_plugins/metadata_markers.py [hunk 3]
```diff
@@ -56,6 +90,57 @@ def pytest_configure(config):
...
+    if verifies_issues:
+        if not verifies_marker:
+            log_and_deselect(item, '--verifies-issues')
+            return False
+        if isinstance(verifies_issues, list):
+            verifies_args = verifies_marker.args[0]
+            if all(issue not in verifies_issues for issue in verifies_args):
+                log_and_deselect(item, '--verifies-issues')
+            return False
+    return True
```
Comment (omkarkhatavkar) [inline pytest_plugins/metadata_markers.py:119]: may be  reduce the nesting of if statements

**Review by pondrejk — APPROVED** [review]

File: pytest_plugins/metadata_markers.py (hunk 3 above)
Comment (jameerpathan111) [reply pytest_plugins/metadata_markers.py:119]: Tried my best with that already😅 did quite brainstorming to get to this while keeping all the scenarios in mind😄

**Review by JacobCallahan — APPROVED** [review]
ACK. Merge pending actions from @omkarkhatavkar's feedback.

File: pytest_plugins/metadata_markers.py (hunk 2 above)
Comment (jameerpathan111) [reply pytest_plugins/metadata_markers.py]: Updated the PR:)

**Review by omkarkhatavkar — APPROVED** [review]

**Pull request merged**